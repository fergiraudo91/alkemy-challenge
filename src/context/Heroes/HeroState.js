import React, { useReducer, useState } from 'react';
import HeroContext from './HeroContext';
import HeroReducer from './HeroReducer';
const HeroState = (props) => {

    const initialState = {
        heroes: [],
        good: 0,
        evil: 0
    }

    const [state, dispatch] = useReducer(HeroReducer, initialState);

    const addGood = () => {
        dispatch({
            type: "ADD_GOOD",
            payload: 1
        })
    }

    const addEvil = () => {
        dispatch({
            type: "ADD_EVIL",
            payload: 1
        })
    }

    const addHeroes = (hero) => {
        dispatch({
            type: 'ADD_HERO',
            payload: hero
        })
    }

    const deleteHeroes = () => {

    }

    return (
        <HeroContext.Provider value={{
            heroes: state.heroes,
            good: state.good,
            evil: state.evil,
            addHeroes,
            deleteHeroes,
            addGood,
            addEvil
        }}>
            {props.children}
        </HeroContext.Provider>
    )
}

export default HeroState;