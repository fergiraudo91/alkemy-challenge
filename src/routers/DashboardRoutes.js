import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'
import { HeroScreen } from '../components/heroes/HeroScreen'
import { HomeScreen } from '../pages/HomeScreen'
import { Navbar } from '../components/ui/Navbar';
import { SearchScreen } from '../pages/SearchScreen';

export const DashboardRoutes = () => {
    
    return (
        <>
            <Navbar />
                <Switch>
                    <Route exact path="/home" component={ HomeScreen } />
                    <Route exact path="/hero/:heroeId" component={ HeroScreen } />
                    <Route exact path="/search" component= { SearchScreen } />
                    <Redirect to="/home" />
                </Switch>
        </>
    )
}
