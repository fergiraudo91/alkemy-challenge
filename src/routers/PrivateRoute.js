import React from 'react';
import { Redirect, Route } from 'react-router-dom';
import PropTypes from 'prop-types';

export const PrivateRoute = ({
    isAuthenticated,
    component: Component,
    ...rest
}) => {
    localStorage.setItem('lastPath', rest.location.pathname);
    const token = localStorage.getItem('token');
    return (
       <Route
       component = {(props) => (
           (isAuthenticated && token)
           ? <Component {...props}/>
           : <Redirect to="/login" />
       )}
       {...rest}
       />
    )
}

PrivateRoute.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}
