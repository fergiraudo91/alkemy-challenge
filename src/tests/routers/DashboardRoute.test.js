import React from 'react';
import { mount } from 'enzyme';
import { DashboardRoutes } from '../../routers/DashboardRoutes';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import HeroState from '../../context/Heroes/HeroState';


describe('Testing <DashBoardRoutes />', () => {


    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'Alkemy'
        }
    }

    const wrapper = mount(
        <HeroState>
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        </HeroState>);

    test('should display correctly and show the username', () => {

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-user').text().trim()).toBe(contextValue.user.name);

    });

})
