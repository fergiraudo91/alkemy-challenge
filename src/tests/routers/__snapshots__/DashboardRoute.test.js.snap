// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <DashBoardRoutes /> should display correctly and show the username 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      aria-current="page"
      className="navbar-brand active"
      href="/home"
      onClick={[Function]}
      style={Object {}}
    >
      Home 
      <i
        className="fas fa-home"
      />
    </a>
    <button
      aria-controls="navbarNavAltMarkup"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarNavAltMarkup"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse justify-content-between"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search Hero 
          <i
            className="fas fa-search"
          />
        </a>
      </div>
      <div
        className="navbar-nav"
      >
        <span
          className="nav-link nav-item text-info text-user"
        >
          Alkemy
        </span>
        <button
          className="nav-link ml-auto align-content-end mr-3 btn btn-dark"
          onClick={[Function]}
          to="/logout"
        >
          Logout 
          <i
            className="fas fa-sign-out-alt"
          />
        </button>
      </div>
    </div>
  </nav>,
  <div
    className="main"
  >
    <h1>
      My Team
    </h1>
    <div
      className="empty-hero animate__animated animate__fadeIn"
    >
      <h5>
        Your team is empty
      </h5>
      <p>
        Please search a hero and add it to your team
      </p>
      <a
        className="btn btn-warning"
        href="/search"
        onClick={[Function]}
      >
        Search Hero
      </a>
    </div>
  </div>,
]
`;
